core/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/elasticapi/ElasticExtensions.kt

```diff
@@ -145,11 +145,14 @@ fun XContentBuilder.optionalUserField(name: String, user: User?): XContentBuilde
 }
 
 fun addFilter(user: User, searchSourceBuilder: SearchSourceBuilder, fieldName: String) {
-    val filterBckendRoles = QueryBuilders.termsQuery(fieldName, user.backendRoles)
-    val queryBuilder = searchSourceBuilder.query() as BoolQueryBuilder
-    searchSourceBuilder.query(queryBuilder.filter(filterBckendRoles))
+    addUserRolesFilter(user.backendRoles, searchSourceBuilder, fieldName)
 }
 
+fun addUserRolesFilter(userRoles: List<String>, searchSourceBuilder: SearchSourceBuilder, fieldName: String) {
+    val filterBackendRoles = QueryBuilders.termsQuery(fieldName, userRoles)
+    val queryBuilder = searchSourceBuilder.query() as BoolQueryBuilder
+    searchSourceBuilder.query(queryBuilder.filter(filterBackendRoles))
+}
```

Comment:
Why do u need to create a new function? If it is because of the function name, I would then rather rename.

Reply:
This change is from my previous commit, in that commit, I need this function. Will remove this one to make the change minimal .